evcouplings/complex/protocol.py

```diff
@@ -111,27 +315,373 @@ def _modify_segments(seg_list, seg_prefix):
+        kwargs["first_annotation_file"],
+        kwargs["second_annotation_file"],
+        kwargs["first_genome_location_file"],
+        kwargs["second_genome_location_file"],
+        outcfg["concatentation_statistics_file"]
+    )
+
+    return outcfg
+
+
+def best_hit(**kwargs):
+    """
+    Protocol:
+
+    Concatenate alignments based on the best hit 
+    to the focus sequence in each species
+
+    Parameters
+    ----------
+    Mandatory kwargs arguments:
+        See list below in code where calling check_required
+        (TODO: explain meaning of parameters in detail).
+
+    Returns
+    -------
+    outcfg : dict
+        Output configuration of the pipeline, including
+        the following fields:
+
+        alignment_file
+        raw_alignment_file
+        focus_mode
+        focus_sequence
+        segments
+        frequencies_file
+        identities_file
+        num_sequences
+        num_sites
+        raw_focus_alignment_file
+        statistics_file
+    """
+    check_required(
+        kwargs,
+        [
+            "prefix",
+            "first_alignment_file", "second_alignment_file",
+            "first_focus_sequence", "second_focus_sequence",
+            "first_focus_mode", "second_focus_mode",
+            "first_segments", "second_segments",
+            "first_identities_file", "second_identities_file",
+            "first_annotation_file", "second_annotation_file"
+        ]
+    )
+
+    prefix = kwargs["prefix"]
+
+    # make sure input alignments
+    verify_resources(
+        "Input alignment does not exist",
+        kwargs["first_alignment_file"], kwargs["second_alignment_file"]
+    )
 
-    outcfg = {
-        "alignment_file": None,  # TODO: specify
-        "focus_mode": True,
-        "focus_sequence": None,  # TODO: specify
-        "segments": [s.to_list() for s in segments_complex],
-        # optional but good to have:
-        "num_sites": None,
-        "num_sequences": None,
-        # "effective_sequences": n_eff # TODO: could compute this like in align stage
-        # TODO: there are more outputs that we could add here (not mandatory),
-        # e.g. single column frequencies in concatenated alignment
-    }
+    # make sure output directory exists
+    create_prefix_folders(prefix)
+
+    def _load_monomer_info(annotations_file,
+                           identities_file,
+                           target_sequence,
+                           alignment_file):
+
+        # read in annotation to a file and rename the appropriate column
+        annotation_table = read_species_annotation_table(annotations_file)
+
+        # This dictionary maps identifiers used to calculate pairing
+        # to identifiers used to index into the sequence alignment
+        # Make sure the target sequence is included in this dictionary
+        # TODO: fix this so that we don"t assume target sequence is the first sequence
+        id_to_header = {x: [x] for x in annotation_table.id}
```

Comment:
not sure if I fully follow what the point of this and the next line is? why is this dictionary necessary if it just maps from self to self?